Sources/Hummingbird/Server/RequestID.swift

```diff
@@ -0,0 +1,40 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Hummingbird server framework project
+//
+// Copyright (c) 2023 the Hummingbird authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See hummingbird/CONTRIBUTORS.txt for the list of Hummingbird authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Atomics
+
+/// Generate Unique ID for each request
+struct RequestID: CustomStringConvertible {
+    let low: UInt64
+
+    init() {
+        self.low = Self.globalRequestID.loadThenWrappingIncrement(by: 1, ordering: .relaxed)
+    }
+
+    var description: String {
+        String(Self.high, radix: 16) + self.formatAsHexWithLeadingZeros(self.low)
+    }
```

Comment:
* It's certainly cleaner this way, but it still feels unnecessary to transform the integer to a `String` once per `RequestID.description` call, considering `.description` will be common due to the logging statement.
* Doesn't this hex description (partially) defeat the point of having a nice easily-human-readable integer as the request's id? At that point a random UUID could just be the simpler solution although i assume it's somewhat more costly to create.

Reply:
Good point about transforming `high` repeatedly. 

I'm not using UUID because one it requires Foundation and two it is slower to generate. I reduced it to hex to shorten the id and the first 16 chars end up being a process id. Anyway not sure a 128bit integer is human readable.

Reply:
Hmm fair enough. I now think it's fine being a hex 🤔 i was thinking someone might rather read/have it as a decimal number, but I can't come up with enough reasons why that should be important. They can always turn the hex back to an integer trivially if it's that important 🤔